Tasks/UseDotNetV2/usedotnet.ts

```diff
@@ -2,52 +2,88 @@
+    if (useGlobalJson && packageType == "sdk") {
+        let globalJsonFetcherInstance = new globalJsonFetcher(workingDirectory);
+        let versionsToInstall: VersionInfo[] = await globalJsonFetcherInstance.GetVersions(packageType);
+        versionsToInstall = versionsToInstall.filter(d => !dotNetCoreInstaller.isVersionInstalled(d.getVersion()));
+        for (let index = 0; index < versionsToInstall.length; index++) {
+            const version = versionsToInstall[index];
+            let url = versionFetcher.getDownloadUrl(version);
+            if (!dotNetCoreInstaller.isVersionInstalled(version.getVersion())) {
+                await dotNetCoreInstaller.downloadAndInstall(version, url);
+            }
+        }
+    } else if (versionSpec) {
+        console.log(tl.loc("ToolToInstall", packageType, versionSpec));
+        let versionSpecParts = new VersionParts(versionSpec);
         let versionInfo: VersionInfo = await versionFetcher.getVersionInfo(versionSpecParts.versionSpec, packageType, includePreviewVersions);
         if (!versionInfo) {
             throw tl.loc("MatchingVersionNotFound", versionSpecParts.versionSpec);
         }
-
-        let dotNetCoreInstaller = new VersionInstaller(packageType, installationPath);
         if (!dotNetCoreInstaller.isVersionInstalled(versionInfo.getVersion())) {
             await dotNetCoreInstaller.downloadAndInstall(versionInfo, versionFetcher.getDownloadUrl(versionInfo));
         }
-
-        tl.prependPath(installationPath);
-
-        // Set DOTNET_ROOT for dotnet core Apphost to find runtime since it is installed to a non well-known location.
-        tl.setVariable('DOTNET_ROOT', installationPath);
-
-        // By default disable Multi Level Lookup unless user wants it enabled.
-        let performMultiLevelLookup = tl.getBoolInput("performMultiLevelLookup", false);
-        tl.setVariable("DOTNET_MULTILEVEL_LOOKUP", !performMultiLevelLookup ? "0" : "1");
+    } else {
+        throw new error("Hey developer you have called the method `installDotNet` without a `versionSpec` or without `useGlobalJson`. that is impossible.");
     }
 
-    // Install NuGet version specified by user or 4.4.1 in case none is specified
-    // Also sets up the proxy configuration settings.
-    const nugetVersion = tl.getInput('nugetVersion') || '4.4.1';
-    await NuGetInstaller.installNuGet(nugetVersion);
-
+    tl.prependPath(installationPath);
+    // Set DOTNET_ROOT for dotnet core Apphost to find runtime since it is installed to a non well-known location.
+    tl.setVariable('DOTNET_ROOT', installationPath);
+    // By default disable Multi Level Lookup unless user wants it enabled.
+    tl.setVariable("DOTNET_MULTILEVEL_LOOKUP", !performMultiLevelLookup ? "0" : "1");
     // Add dot net tools path to "PATH" environment variables, so that tools can be used directly.
     addDotNetCoreToolPath();
```

Comment:
This should be done irrespective of a version being installed or not, and thus should be moved to `run`